dnf/comps.py

```diff
@@ -595,6 +595,9 @@ def _removable_grp(self, group_id):
     def _environment_install(self, env_id, pkg_types, exclude, strict=True, exclude_groups=None):
         assert dnf.util.is_string_type(env_id)
         comps_env = self.comps._environment_by_id(env_id)
+        if not comps_env:
+            raise ValueError(_("Environment id '%s' does not exist.") % ucd(env_id))
```

Comment:
Why do you  want to use ValueError()?

Reply:
I've used the same error class that is used in `_group_install()`.

Reply:
Unfortunately it will change API behaviour. In past we have also a problem with ValueErrors and traceback, therefore we replaced it by a dnf.error.

Reply:
You can check it in dnf/comp.py history.

Reply:
Could you be more concrete (and helpful) with you suggestions, please? I went to look into the history and I can only assume you meant this commit: https://github.com/rpm-software-management/dnf/commit/031322e670c70446c27f5e5ad7d20140323dcd87

Which mentions `ValueError` but doesn't change anything about the exceptions or their handling, it fixes the problem in a different way. So maybe you've meant something else?

Reply:
Probably I overlooked something, but I see that API function base.environment_install() will now raise ValueError in case that provided id is not known, but before your change it returned 0 for the same situation.

Reply:
I don't think so, I think this will actually crash on line 601 of this file, since `comps_env` will be `None` (looking at this, the check on line 598 should be `if comps_env is not None:`).

I have fixed multiple errors similar to this one and all of those were crashing in similar manner.

Reply:
What about to replace ValueError by dnf.comps.CompsError. Then it will work like expected. Method install_or_skip will do what is expected, install or skip.

Reply:
Ok. I'll have a look and try to replace the other occurrences of `VauleError` then, to make it consistent?

Reply:
Well, that actually doesn't work, because `CompsError` is caught and logged in `install_or_skip()`, so if I change the exception class to `CompsError`, the result is I'm not getting any exception at all from either `base.group_install()` or `base.environment_install()`.

Reply:
The original code returned 0 when no comps transaction and in other cases >=1. Therefore information was there.

Reply:
I need an actual error (with a message), not just an information whether something was installed or not. The behavior is just wrong from the API perspective as well, but since we don't want to break compatibility, I'm inclined to just keep it as it is - otherwise bigger changes would be needed to make it work well.

Reply:
In DNF-5 the code will be moved into C++ therefore it can be improved.

Reply:
@lukash What's the plane for ValueError?